backend/src/users/routers.py

```diff
@@ -20,21 +24,33 @@ async def login_for_access_token(
         form_data: OAuth2PasswordRequestForm = Depends(), database: Session = Depends(session_opener)
 ):
     """Login and verify user"""
+    logging.debug(f"Login initialised: {form_data.username}")
     user = check_user_password_is_correct(database, form_data.username, form_data.password)
-    access_token = create_access_token(
-        data={"sub": str(user.username)}, expires_delta=timedelta(minutes=30)
-    )
+    if not user:
+        logging.debug(f"Failed to login: {form_data.username}")
+        raise HTTPException(status_code=400, detail="Incorrect username or password")
+    try:
+        access_token = create_access_token(
+            data={"sub": str(user.username)}, expires_delta=timedelta(minutes=30)
+        )
+    except Exception as e:
+        capture_exception(e)
+        return HTTPException(status_code=400, detail="Something went wrong while processing the request")
+    logging.debug(f"Logged in: {form_data.username}")
     return {"access_token": access_token, "token_type": "bearer"}
 
 @router.post("/register")
 def create_new_user(user: UserAuthSchema, database: Session = Depends(session_opener)):
+    logging.debug(f"Creating user: {user.username}")
     hashed_password = password_context.hash(user.password)
     database_user = User(username=user.username, hashed_password=hashed_password)
     database.add(database_user)
     database.commit()
```

Comment:
`Session.commit() `is possibly throwing exception, should add a try..except expression to handle exception